spec/dapp/chains.ts

```diff
@@ -0,0 +1,113 @@
+import groupBy from 'lodash.groupby';
+
+export enum Names {
+  Eximchain = 'Eximchain',
+  // Commented out until we have a definite txExecutor URL
+  // EximGamma = 'EximchainGamma',
+  Ethereum = 'Ethereum',
+  Ropsten = 'Ropsten',
+  Kovan = 'Kovan',
+  Rinkeby = 'Rinkeby',
+  Goerli = 'Goerli'
+}
+
+export interface Details {
+  id: number
+  genesisHash: string
+  web3Url: string
+  name: Names
+  displayName: string
+}
+
+export function isName(val: string): val is Names {
+  return Object.values(Names).includes(val as Names);
+}
+
+export function detailsFromName(name: string): Details | null {
+  let details = detailsByName()[name];
+  return details ? details : null;
+}
+
+export function detailsFromUrl(url:string): Details | null {
+  let detail = allDetails().find(each => each.web3Url === url);
+  return detail ? detail : null;
+}
+
+export function detailsById(): { [key: number] : Details[] } {
+  return groupBy(allDetails(), detail => detail.id);
+}
+
+export function detailsByName(): { [key: string]: Details } {
+  return {
+    [Names.Eximchain]: Eximchain(),
+    [Names.Ethereum]: Ethereum(),
+    [Names.Goerli]: Goerli(),
+    [Names.Kovan]: Kovan(),
+    [Names.Rinkeby]: Rinkeby(),
+    [Names.Ropsten]: Ropsten()
+  }
+}
+
+export function allDetails(): Details[] {
+  return Object.values(detailsByName);
+}
+
+export function Eximchain(): Details {
+  return {
+    id: 1,
+    genesisHash: '0x722138a9f3635c65747a8e2eeac7e7963846fb952b8931da257395fd7656c3dd',
+    web3Url: "https://tx-executor-stress-test.eximchain.com",
```

Comment:
I sort of think we should use `tx-executor-us-east.eximchain.com` instead.  This also makes me wonder how much of a change it would be to support multiple possible Web3 URLs in the future.  We actually have 3 public tx executors for main net

Reply:
The wrinkle here is that the Web3URL is the only way we have to determine which network a Dapp is deployed on.  If we change that data, then we also need to migrate all the dapps which have those URLs

Reply:
Resolution is that we're leaving it as is & creating some tech debt to eventually support multiple Web3URLs for the same network.